.github/workflows/test-deptrac.yml

```diff
@@ -0,0 +1,82 @@
+# When a PR is opened or a push is made, perform an
+# architectural inspection on the code using Deptrac.
+name: Deptrac
+
+on:
+  pull_request:
+    branches:
+      - 'develop'
+      - '4.*'
+    paths:
+      - 'app/**'
+      - 'system/**'
+      - 'composer.json'
+      - 'depfile.yaml'
+      - '.github/workflows/test-deptrac.yml'
+  push:
+    branches:
+      - 'develop'
+      - '4.*'
+    paths:
+      - 'app/**'
+      - 'system/**'
+      - 'composer.json'
+      - 'depfile.yaml'
+      - '.github/workflows/test-deptrac.yml'
+
+jobs:
+  build:
+    name: PHP ${{ matrix.php-versions }} Architectural Inspection
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        php-versions: ['7.4', '8.0']
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-versions }}
+          tools: composer, pecl, phive, phpunit
+          extensions: intl, json, mbstring, gd, mysqlnd, xdebug, xml, sqlite3
+
+      - name: Use latest Composer
```

Comment:
No need for this as setup-php installs the latest available.

Reply:
Ironically, I took this from our PHPStan workflow! Never used it myself but I figured somebody knew something I didn't :D

https://github.com/codeigniter4/CodeIgniter4/blob/1928006ccc9fa86be11bb8ee5708f4d7bec95fda/.github/workflows/test-phpstan.yml#L43-L44

Reply:
I think this was because composer v2 wasn't released then and anyone can force update with `composer self-update --2`